### ISSUE TITLE
Shipping zones are using the billing address and not the shipping address

### ISSUE THREAD

**Issue by wvassello** [woocommerce/woocommerce#25812]
**Description**

I have created two shipping zones. One for the continental US and the other for Alaska & Hawaii. When the billing address state is set to the continental US, that zone is selected. When the billing address state is Alaska or Hawaii, the A&H zone is selected. The shipping state has no effect on the selected shipping zone.

Shipping options to force shipping to billing address and default to billing address are not selected.

This effects the woocommerce advanced free shipping plugin as well. So it seems to be core functionality.

**To Reproduce**
1. Create two shipping zones that are controlled by different US states.
2. In the checkout, change the billing state to the different states and watch the shipping selection change.
3. Change the shipping state in the checkout and see it has no effect on the shipping selection.

**Screenshots**

**Expected behavior**
Shipping zones should be selected based on shipping address values and not billing address values.

**Isolating the problem (mark completed items with an [x]):**
- [X ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
Wordpress - Version 5.3.2
Woocommerce Version 3.9.2
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
**Note
You will see an advanced free shipping plugin currently installed. This bug occurs with and without this plugin. So the bug is overriding this plugin as well as the core woocommerce code.

### WordPress Environment ###

WordPress address (URL): http://test.mymedicalimages.com
Site address (URL): http://test.mymedicalimages.com
WC Version: 3.9.2
REST API Version: ✔ 1.0.7
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 384 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.1.33 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 256 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.68.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.7.29
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔
Zapier Integration - Active Feeds: ✔ 1 active Zapier Feed.
Zapier Integration - Cron Tasks: ✔ 0 pending Zapier cron tasks.
Zapier Feed #1 Name: ✔ The Box
Zapier Feed #1 Webhook URL: ✔ https://hooks.zapier.com/hooks/catch/6663053/odgm7fy/
Zapier Feed #1 Trigger: ✔ New Order
Zapier Feed #1 Test Result: ✔ Successfully sent sample data to https://hooks.zapier.com/hooks/catch/6663053/odgm7fy/.

### Database ###

WC Database Version: 3.9.2
WC Database Prefix: wp_
Total Database Size: 25.24MB
Database Data Size: 22.83MB
Database Index Size: 2.41MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_itemmeta: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_commentmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_comments: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_em_modals: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_em_modal_metas: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_em_themes: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_em_theme_metas: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_content_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_core_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_flags: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_languages: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_icl_languages_translations: Data: 0.10MB + Index: 0.10MB + Engine MyISAM
wp_icl_locale_map: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_message_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_mo_files_domains: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_node: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_reminders: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_strings: Data: 1.97MB + Index: 1.39MB + Engine MyISAM
wp_icl_string_packages: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_string_pages: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_string_positions: Data: 0.10MB + Index: 0.04MB + Engine MyISAM
wp_icl_string_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_string_translations: Data: 0.96MB + Index: 0.30MB + Engine MyISAM
wp_icl_string_urls: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_translate: Data: 0.09MB + Index: 0.00MB + Engine MyISAM
wp_icl_translate_job: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_translations: Data: 0.01MB + Index: 0.03MB + Engine MyISAM
wp_icl_translation_batches: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_icl_translation_status: Data: 0.10MB + Index: 0.01MB + Engine MyISAM
wp_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_options: Data: 1.95MB + Index: 0.06MB + Engine MyISAM
wp_postmeta: Data: 2.37MB + Index: 0.18MB + Engine MyISAM
wp_posts: Data: 14.90MB + Index: 0.16MB + Engine MyISAM
wp_smush_dir_images: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_termmeta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_terms: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_term_relationships: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_term_taxonomy: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_usermeta: Data: 0.04MB + Index: 0.02MB + Engine MyISAM
wp_users: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_product_meta_lookup: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpgmza: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_categories: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_category_maps: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_circles: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_maps: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_polygon: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_polylines: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpgmza_rectangles: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_yoast_seo_meta: Data: 0.02MB + Index: 0.03MB + Engine MyISAM

### Post Type Counts ###

attachment: 660
customize_changeset: 5
custom_css: 2
et_pb_layout: 44
hm_custom_css: 5
hm_custom_javascript: 2
nav_menu_item: 14
oembed_cache: 4
page: 26
post: 5
postman_sent_mail: 32
product: 3
product_variation: 12
revision: 1007
scheduled-action: 6
shop_order: 11
testimonial: 12
testimonial_rotator: 3
tt_font_control: 1
wafs: 3
wc_zapier_feed: 1
wpforms: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (21) ###

404page - your smart custom 404 error page: by Peter Raschendorfer – 11.2.1
Akismet Anti-Spam: by Automattic – 4.1.3
All Meta Tags: by Space X-Chimp – 4.38
Custom Order Numbers for WooCommerce: by Tyche Softwares – 1.2.9 – Not tested with the active version of WooCommerce
DCO Insert Analytics Code: by Denis Yanchevskiy – 1.1.3
Divi Carousel: by DiviGear – 2.0.8
Easy Google Fonts: by Titanium Themes – 1.4.4
Freshchat: by Freshchat – 2.3.3
GTranslate: by Translate AI Multilingual Solutions – 2.8.51
Show Private: by Amir Masoud Irani( AMIB ) – 0.2.1
Testimonial Rotator: by Hal Gatewood – 3.0.1
Typekit Fonts for WordPress: by OM4 – 1.9.0
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.16.22

Checkout Field Editor for WooCommerce: by ThemeHigh – 1.4.1
WooCommerce Advanced Free Shipping: by Jeroen Sormani – 1.1.4 – Not tested with the active version of WooCommerce
WooCommerce Stripe Gateway: by WooCommerce – 4.3.2
WooCommerce Zapier Integration: by OM4 – 1.9.6 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.9.2
WP Performance Score Booster: by Dipak C. Gajjar – 1.9.2.2
Smush: by WPMU DEV – 3.4.2
WP Super Cache: by Automattic – 1.7.1

### Inactive Plugins (13) ###

Custom CSS and Javascript: by Potent Plugins – 2.0.9
Divi Tweaker: by Divi Web Design – 1.1.3
Duplicate Page: by mndpsingh287 – 4.0
Easy Modal: by Daniel Iser – 2.1.0
Google Analytics Dashboard for WP (GADWP): by ExactMetrics – 5.3.10
Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 7.10.3
Google XML Sitemaps: by Arne Brachhold – 4.1.0
Insert Headers and Footers: by WPBeginner – 1.4.4
Post SMTP: by Yehuda Hassine – 2.0.10
Speed Booster Pack: by Optimocha – 3.8.4.3
W3 Total Cache: by Frederick Townes – 0.9.7.3
WP Google Analytics Events: by PineWise – 2.6.0
Yoast SEO: by Team Yoast – 12.9.1

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Must Use Plugins (1) ###

ET Support Center :: Must-Use Plugins Autoloader: by Elegant Themes –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #2820 - /starter-kit/
Cart: #4862 - /cart/
Checkout: #4863 - /checkout/
My account: #4864 - /my-account/
Terms and conditions: #433 - /terms-of-service/

### Theme ###

Name: candyland
Version: 1.0.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Divi
Parent Theme Version: 4.3.4
Parent Theme Author URL: http://www.elegantthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 6
Oldest: 2020-02-07 20:54:21 +0000
Newest: 2020-03-03 21:36:07 +0000

Pending: 0
Oldest: –
Newest: –

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`

</details>

**Comment by wvassello** [comment]
This is a pressing issue and is preventing our company from deploying our commerce pages.
Any help or escalation of this issue is extremely appreciated.
-wvassello

**Comment by wvassello** [comment]
This is a pressing issue and is preventing our company from deploying our commerce pages.
Any help or escalation of this issue is extremely appreciated.
-wvassello

**Comment by tammullen** [comment]
Hi @wvassello 

Thank you for submitting the issue. However, I can’t reproduce the issue you reported using the steps you provided. Everything is working as expected on my end. 


![Screenshot_2020-03-05_at_10_58_34](https://user-images.githubusercontent.com/24649833/75977436-4f049700-5ed4-11ea-88e9-2624ef25d430.png)
I set up two state specific rates
![Screenshot_2020-03-05_at_11_00_32](https://user-images.githubusercontent.com/24649833/75977431-4dd36a00-5ed4-11ea-9e30-34b40bc3a4e0.png)
Set Shipping options to use shipping address
![screencapture-corporate-newt-jurassic-ninja-checkout-2020-03-05-11_04_02](https://user-images.githubusercontent.com/24649833/75977417-47dd8900-5ed4-11ea-8885-8b258794b31d.png)
When `Ship to a different address?` is unticked the correct rate is displayed for the state selected.
![Screenshot_2020-03-05_at_11_03_52](https://user-images.githubusercontent.com/24649833/75977426-4ad87980-5ed4-11ea-8dc5-4c1efa99103a.png)
When `Ship to a different address?` is ticked the correct rate is displayed for the state in the shipping address


![Screenshot_2020-03-05_at_11_12_11](https://user-images.githubusercontent.com/24649833/75977410-4318d500-5ed4-11ea-9778-acd2874cf257.png)
I also tried using the zones in your example and it still worked

Note: the order of shipping zones is very important when you have multiple zones that can include the same location. In this example Hawaii would match for both shipping zones. Zones match from top to bottom so in this example Hawaii would match the top zone of A&H.

See here for more details
https://docs.woocommerce.com/document/setting-up-shipping-zones/#section-4

Can you confirm you have your zones ordered correctly or any other details that may help us evaluate this issue further?
​

**Comment by wvassello** [comment]
Thank you tanmullen for looking into this. I can confirm that the shipping zones are not in conflict because I did not overlap them. My zones are state specific (I did not use the general US filter) and list each state individually. Is there a reason I should consider overlapping them?

I do have a block of code forcing the use of a shipping address (see below), but I have removed this code and still have the same behavior. But if anything, this should prohibit the engine from using the billing address.

Like I described though, the behavior is that the shipping switches just fine and is acting as expected if the billing state is changed, but changing the shipping state has no effect. I confirm this in debug mode to watch the zone selection.

What other info can I provide or place I can look for the definitions of the billing and shipping variables. Maybe these are switched someplace.


/**
 * Hide the 'Ship to different address' checkbox / add custom heading.
 * If wanted you can replace the text in here. 
 */
function ace_js_force_shipping_address() {
    ?><script type="text/javascript">
        var diffAddress = document.querySelector('#ship-to-different-address');
        diffAddress.outerHTML = '<h3>Shipping details</h3><input type="checkbox" checked="checked" value="1" name="ship_to_different_address" class="hidden" style="display: none;" />';
    </script><?php
}
add_action( 'wp_footer', 'ace_js_force_shipping_address' );

/**
 * Force shipping address on checkout validation
 */
function ace_force_posted_data_ship_to_different_address( $posted_data ) {
    $posted_data['ship_to_different_address'] = true;

    return $posted_data;
}
add_filter( 'woocommerce_checkout_posted_data', 'ace_force_posted_data_ship_to_different_address' );

// Note that this overrides the 'Shipping destination' option in the Woo settings
add_filter( 'woocommerce_ship_to_different_address_checked',  '__return_true' );

// If you have the possibility of virtual only orders you may want to comment this out
add_filter( 'woocommerce_cart_needs_shipping_address',  '__return_true' );

// Order always has shipping (even with local pickup for example)
add_filter( 'woocommerce_order_needs_shipping_address',  '__return_true' );

**Comment by wvassello** [comment]
Its really like the "use billing as shipping address" switch is on and will not turn off.

I suppose I could set that value in code instead of relying on the gui radial buttons.

**Comment by wvassello** [comment]
Hello tanmullen,
I solved this issue. It turns out that a plugin that was used to modify the checkout had created a custom field for the shipping state. So therefore, woocommerce no longer referenced it as a true shipping address. 

This custom field was generated when the original intent was to restrict the displayed states for shipping.


Once I removed this custom field and had the system rebuild the default fields, it seems to have cleared it all up.

I was under the impression that another user had removed this plugin and tested. So it possible that even removing the plugin would not have undone the field override.

Thank you for your help and providing a clean check of the issue.
